app/models/server/models/LivechatRooms.js

```diff
@@ -22,6 +22,7 @@ export class LivechatRooms extends Base {
 		this.tryEnsureIndex({ 'v._id': 1 }, { sparse: true });
 		this.tryEnsureIndex({ t: 1, departmentId: 1, closedAt: 1 }, { partialFilterExpression: { closedAt: { $exists: true } } });
 		this.tryEnsureIndex({ source: 1 }, { sparse: true });
+		this.tryEnsureIndex({ 'open': 1, 'v.token': 1 }, { sparse: true });
```

Comment:
looking the existing queries, I think a better index would be:
```
t: 1, 
'v.token': 1,
open: 1,
'email.thread' :1,
```
this could be the only index with `v.token` and it probably fit all queries (except the one at `updateDataByToken` that would require a change to include `t: 'l'` filter as well)

Reply:
The thing with adding "t" as part of the index is that "t" doesn't change, and "t" holds very few possible values. With that, t won't discriminate so much to be part of an index imho 👀 

do you think it's worth it?

(about the "email.thread", yep, I can add it there)

Reply:
my suggestion to add `t` was because I've seen almost all queries using it as well. but yeah, maybe that is not that useful on the index and mongo could already use an index without it.

the overall goal of my suggestion is to have **less** indexes covering **more** queries.. so with change we would end up with 3 indexes with `v.token`:

```
		this.tryEnsureIndex({ 'v.token': 1 }, { sparse: true });
		this.tryEnsureIndex({ 'v.token': 1, 'email.thread': 1 }, { sparse: true });
		this.tryEnsureIndex({ 'open': 1, 'v.token': 1 }, { sparse: true });
```

looking the queries, I think a single index can handle all queries:

```
this.tryEnsureIndex({
    t: 1, // as per messages above, maybe not needed
    'v.token': 1, // the main field of the index (present in all queries)
    'email.thread': 1,
    open: 1, // this last two I'm not sure the "order", need to understand which one happens more
}, { sparse: true });
```